File: js/client_helper.js
const thrift = require('thrift');
const Calculator = require('./types/Calculator');

module.exports = (host, port, path) => {
  const options = {
     transport: thrift.TBufferedTransport,
     protocol: thrift.TBinaryProtocol,
     path: path,
     headers: { "Connection": "close", "Content-Type": "application/x-thrift" },
     https: false
  };
  const connection = thrift.createHttpConnection(host, port, options);
  connection.on('error', err => {
    console.error(err);
  });
  const client = thrift.createHttpClient(Calculator, connection);
  return client;
};

File: ruby/client_helper.rb
$:.push('types')

require 'thrift'
require 'calculator'

module Calculator
  class Client
    def self.from(url, &block)
      transport = Thrift::HTTPClientTransport.new(url)
      protocol = Thrift::BinaryProtocol.new(transport, false, false)

      transport.open
      yield self.new(protocol)
      transport.close
    end
  end
end
